### DISCUSSION TITLE
Slow navigation compared to Xamarin Forms

### DISCUSSION THREAD

**Discussion by pauldiston** [dotnet/maui#14703] [General]
I am in the process of porting a Xamarin Forms application over to .NET MAUI and I am noticing that navigation is quite slow when compared to Xamarin Forms.

For example, I have the following timings for a SimpleView (a series of Labels) :-
_.NET MAUI
[0:] Push:SimpleView took 00:00:01.3732189
[0:] Push:SimpleView took 00:00:01.0152959
[0:] Push:SimpleView took 00:00:00.9931281_

_Xamarin Forms
[0:] Push:SimpleView took 00:00:00.6529151
[0:] Push:SimpleView took 00:00:00.4340805
[0:] Push:SimpleView took 00:00:00.4439873_

The timings get worse with a more complex page (ListView with 3 items, data template selectors, quite a few bindings, etc) :-
_.NET MAUI
[0:] Push:ComplexView took 00:00:05.9756774
[0:] Push:ComplexView took 00:00:04.8321169
[0:] Push:ComplexView took 00:00:04.4887956_

_Xamarin Forms
[0:] Push:ComplexView took 00:00:01.8588404
[0:] Push:ComplexView took 00:00:01.0497019
[0:] Push:ComplexView took 00:00:01.0239016_

As you can see, it takes nearly 5 times as long to navigate to the ComplexView on .NET MAUI, compared to Xamarin Forms.

I am using a NavigationPage with a call to PushAsync. I am also using version 7.0.81.

Is there anything I am missing in terms of migration steps which would help?

**Comment by DDHSchmidt** [comment]
What target OS/device did you use to measure those times?
And if you have measured multiple ones: Do they align or is there a notable outlier?

**Reply by pauldiston** [reply]
This was on an Android 9 device (Samsung S8). The app only targets Android at the moment so this has been my primary focus. I have not looked at other devices or OSes.

**Comment by sjordanGSS** [comment]
Are you compiling in debug or release mode? I've found that in debug mode, MAUI is slower than Xamarin.Forms, but much faster in release.

**Reply by pauldiston** [reply]
I am running in Debug for both .NET MAUI and Xamarin Forms. I would be reluctant to put this down to a Release/Debug issue, Xamarin Forms would be quicker in Release mode so the slowness of .NET MAUI would still be an issue.

**Reply by sjordanGSS** [reply]
You should really be using release mode when judging the speed/responsiveness of software: that's what your users will be seeing after all. Debug mode adds a lot of cruft that slows down the program but is necessary for debugging, as well as just not being as heavily optimised by the compiler.

**Reply by pauldiston** [reply]
I definitely agree with this however I would think you should be able to compare 2 Debug builds without having to produce Release builds.
